packages/@aws-cdk/aws-appsync/lib/graphqlapi.ts

```diff
@@ -150,6 +159,38 @@ export interface OpenIdConnectConfig {
   readonly oidcProvider: string;
 }
 
+/**
+ * Configuration for Lambda authorization in AppSync
+ */
+export interface lambdaAuthorizerConfig {
+  /**
+   * The ARN for the authorizer lambda function. This may be a standard Lambda ARN, a version ARN (.../v3) or alias ARN.
+   * Note: This Lambda function must have the following resource-based policy assigned to it.
+   * When configuring Lambda authorizers in the console, this is done for you.
+   * To do so with the AWS CLI, run the following:
+   *
+   * `aws lambda add-permission --function-name "arn:aws:lambda:us-east-2:111122223333:function:my-function" --statement-id "appsync" --principal appsync.amazonaws.com --action lambda:InvokeFunction`
+   *
+   * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-appsync-graphqlapi-lambdaauthorizerconfig.html
+   */
+  readonly functionArn: string;
```

Comment:
Yes, it's even in the [design guidelines][1]:

> Use strong types (and specifically, construct interfaces) instead of physical attributes when referencing other resources. For example, instead of keyArn, use kms.IKey [awslint:props-no-arn-refs].

[1]: https://github.com/aws/aws-cdk/blob/master/docs/DESIGN_GUIDELINES.md#types

Reply:
Ah I see. I'll reread the design guidelines and change this up to be `IFunction`.

Reply:
@otaviomacedo, I made the change, so this should be ready for primetime whenever you have a chance.